src/ui/src/shape_list_widget.cpp

```diff
@@ -1,10 +1,172 @@
 #include "shape_list_widget.h"
 
-#include "ui_shape_list_widget.h"
+#include <QStringList>
+#include <QFile>
+
 
 ShapeListWidget::ShapeListWidget(QWidget* parent)
-		: QDockWidget(parent), ui_(new Ui::ShapeListWidget) {
-	ui_->setupUi(this);
+		: QDockWidget(parent) {
+
+	// example with checkboxes https://stackoverflow.com/questions/14158191/qt-qtreeview-and-custom-model-with-checkbox-columns
+	
+	model_ = new TreeModel(this);
+	view_ = new QTreeView(this);
+	view_->setModel(model_);
+	view_->setWindowTitle(QObject::tr("Simple Tree Model"));
+	setWidget(view_);
+
+	Test();
+}
+
+void ShapeListWidget::Test() {
+	model_->TestFillWithTxtFile(
+			"C:\\Users\\winroot\\Desktop\\overheat\\local_docs\\default.txt");
+}
+
+// TreeItem
+
+
+TreeItem::TreeItem(const QVector<QVariant>& data, TreeItem* parent)
+		: m_itemData(data), m_parentItem(parent) {}
+
+TreeItem* TreeItem::child(int row) {
+	if (row < 0 || row >= m_childItems.size())
+		return nullptr;
+	return m_childItems.at(row);
+}
+
+QVariant TreeItem::data(int column) const {
+	if (column < 0 || column >= m_itemData.size())
+		return QVariant();
+	return m_itemData.at(column);
+}
+
+int TreeItem::row() const {
+	if (m_parentItem)
+		return m_parentItem->m_childItems.indexOf(const_cast<TreeItem*>(this));
```

Comment:
No need for `const_cast` (at least on clang and gcc).

Narrowing conversion from `qsizetype` (aka `long long`) to signed type `int`. Avoid such conversions or make them explicit